### ISSUE TITLE
Incorrect annotations on enclosing type type parameters

### ISSUE THREAD

**Issue by smillst** [typetools/checker-framework#3983]
```java
@SuppressWarnings("initialization.fields.uninitialized")
public abstract class EnclosingType<T extends @Nullable Number> {
    abstract class Inner{
        abstract T get();
        abstract void set(T t);
    }
    Inner i;
    T t;
    void setI(Inner i) {
        i.set(this.i.get());  // error here
    }
}
```
I expected no error, but the following was issued:
```
error: [argument.type.incompatible] incompatible argument for parameter t of set.
        i.set(this.i.get());
                        ^
  found   : T[ extends @Initialized @Nullable Number super @Initialized @NonNull Void]
  required: T[ extends @Initialized @NonNull Number super @Initialized @NonNull Void]
```

The explicit annotation on the bound of T is ignored.

**Comment by mernst** [comment]
This code example is a false positive, but it is possible to construct a false negative example too.
